src/main/kotlin/com/cultureamp/eventsourcing/EventTypeResolver.kt

```diff
@@ -0,0 +1,38 @@
+package com.cultureamp.eventsourcing
+
+import com.cultureamp.common.asNestedSealedConcreteClasses
+import kotlin.reflect.KClass
+
+interface EventTypeResolver {
+    fun serialize(domainEventClass: Class<out DomainEvent>): EventTypeDescription
+    fun deserialize(eventTypeDescription: EventTypeDescription): Class<out DomainEvent>
+}
+
+object CanonicalNameEventTypeResolver : EventTypeResolver {
+    override fun serialize(domainEventClass: Class<out DomainEvent>) = EventTypeDescription(domainEventClass.canonicalName, null)
+    override fun deserialize(eventTypeDescription: EventTypeDescription) = eventTypeDescription.eventType.asClass<DomainEvent>()!!
+}
+
+class PackageRemovingEventTypeResolver(aggregateTypeToEventType: Map<KClass<*>, KClass<out DomainEvent>>) : EventTypeResolver {
```

Comment:
Wonder if we should treat aggregate type as a String rather than a class?

Reply:
I'd thought about that but it does make the interface kind of funny - the good thing is that users for the library can provide any reimplementation they want for this, in the case where they're doing something sophisticated with how they wire up their event-store and have aggregate-types that don't match the classname of their aggregate classes

Reply:
Also: We will likely need to do something like this in Skills Coach, which doesn't do package-name resolving, but could benefit from the ability to infer aggregate-types given event classes - so we can just build a custom resolver for Skills Coach